src/passes/OptimizeInstructions.cpp

```diff
@@ -1209,6 +1245,284 @@ struct OptimizeInstructions
+              return left;
+            } else if (auto* leftLeft = left->left->dynCast<Binary>()) {
+              //   ((x ^ -1) + (y ^ -1)) ^ -1   ==>   x - (y ^ -1)
+              auto* constLeftLeftRight = leftLeft->right->dynCast<Const>();
+              if (constLeftLeftRight &&
+                  constLeftLeftRight->value.getInteger() == -1LL) {
+                if (auto* leftRight = left->right->dynCast<Binary>()) {
+                  auto* constLeftRightRight =
+                    leftRight->right->dynCast<Const>();
+                  if (constLeftRightRight &&
+                      constLeftRightRight->value.getInteger() == -1LL) {
+                    return Builder(*getModule())
+                      .makeBinary(Abstract::getBinary(type, Abstract::Sub),
+                                  leftLeft->left,
+                                  leftRight);
+                  }
+                }
+              }
+              if (auto* constLeftLeftRight =
+                    leftLeft->right->dynCast<Const>()) {
+                //   ((x ^ -1) + y) ^ -1   ==>   x - y
+                if (constLeftLeftRight->value.getInteger() == -1LL) {
+                  return Builder(*getModule())
+                    .makeBinary(Abstract::getBinary(type, Abstract::Sub),
+                                leftLeft->left,
+                                left->right);
+                }
+              }
+            } else if (auto* leftRight = left->right->dynCast<Binary>()) {
+              //   (x + (y ^ -1)) ^ -1   ==>   y - x
+              if (auto* constLeftRightRight =
+                    leftRight->right->dynCast<Const>()) {
+                if (constLeftRightRight->value.getInteger() == -1LL) {
+                  return Builder(*getModule())
+                    .makeBinary(Abstract::getBinary(type, Abstract::Sub),
+                                leftRight->left,
+                                left->left);
+                }
+              }
+            }
+          } else if (left->op == Abstract::getBinary(type, Abstract::Xor)) {
+            //   (x ^ -1) ^ -1   ==>   x
+            if (auto* constLeftRigth = left->right->dynCast<Const>()) {
+              if (constLeftRigth->value.getInteger() == -1LL) {
+                return left->left;
+              }
+            }
+          }
+        }
+      }
+    } else if (auto* left = binary->left->dynCast<Binary>()) {
+      //  (x ^ -1) ^ (y ^ -1)   ==>   x ^ y
+      if (auto* right = binary->right->dynCast<Binary>()) {
+        if (auto* constLeftRight = left->right->dynCast<Const>()) {
+          if (constLeftRight->value.getInteger() == -1LL) {
+            if (auto* constRightRight = right->right->dynCast<Const>()) {
+              if (constRightRight->value.getInteger() == -1LL) {
+                binary->left = left->left;
+                binary->right = right->left;
+                return binary;
+              }
+            }
+          }
+        }
+      }
+    }
```

Comment:
@kripken Could you help with this?
fuzzer fail in this branch with seed `17413978523681193292`.

reduced module:
```wat
(module
  (type $t0 (func (param i32) (result i32)))
  (func $f0 (type $t0) (param $p0 i32) (result i32)
    global.get $g0
    f32.convert_i32_s
    f32.const 0x0p+0 (;=0;)
    f32.gt
    f64.const 0x1p+0 (;=1;)
    i32.const 0
    i32.load16_u offset=22
    f64.convert_i32_s
    f64.ne
    i32.xor)
  (memory $M0 1 1)
  (global $g0 (mut i32) (i32.const 65536)))
```
But it's not really help with determine problem.

Reply:
Fixed